test/jdk/java/io/File/libGetXSpace.c

```diff
@@ -124,9 +128,24 @@ Java_GetXSpace_getSpace0
         array[3] = (jlong)freeBytesAvailable.QuadPart;
     }
 #else
+    int len = (int)(*env)->GetStringLength(env, root);
+    char* chars = (char*)malloc((len + 1)*sizeof(char));
+    if (chars == NULL) {
+        (*env)->ReleaseStringChars(env, root, strchars);
+        JNU_ThrowByNameWithLastError(env, "java/lang/RuntimeException",
+                                     "malloc");
+        return JNI_FALSE;
+    }
+
+    for (int i = 0; i < len; i++) {
+        chars[i] = (char)strchars[i];
+    }
```

Comment:
Is this `memcpy(chars, strchars, len);`

Reply:
I don't think so if `chars` is `char *` and `strchars` is `jchar*` as I think `char` is 8 bits and `jchar` is 16 bits.

Reply:
I just wonder if jchar is UTF-16 and char is platform encoding, would it work by simply casting the low 8bit to char?

Reply:
Probably not. This is only intended to cover the constrained case of this test. Something like the macro `WITH_PLATFORM_STRING` in `libjava/io_util.h` would be needed for the general case, but I thought that that might be overdoing it for here. I could of course be mistaken.

Reply:
OK, if the usage is limited and covers the required test cases, that sounds reasonable.

Reply:
Only file system roots are passed as the paths handed to the `getSpace0()` method so hopefully it should be all right.